llvm/lib/Target/RISCV/GISel/RISCVInstructionSelector.cpp

```diff
@@ -164,6 +164,25 @@ bool RISCVInstructionSelector::select(MachineInstr &MI) {
   preISelLower(MI, MIB, MRI);
   const unsigned Opc = MI.getOpcode();
 
+  switch(Opc) {
+  case RISCV::G_RISCV_ADD_LO: {
+    Register DstReg = MI.getOperand(0).getReg();
+    const LLT XLenLLT = LLT::scalar(STI.getXLen());
+    RBI.constrainGenericRegister(DstReg, RISCV::GPRRegClass, MRI);
+    MI.setDesc(TII.get(RISCV::ADDI));
+    MRI.setType(DstReg, XLenLLT);
```

Comment:
Why do we need to call setType on a virtual register that's got a register class?